lib/screens/pets/add-pet/add_pet_form.dart

```diff
@@ -0,0 +1,165 @@
+import 'dart:io';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:pet_app/helpers/app_dialogs.dart';
+import 'package:pet_app/model/pet.dart';
+import 'package:pet_app/services/auth/auth_service.dart';
+import 'package:pet_app/services/services.dart';
+import 'package:pet_app/services/storage/storage_service.dart';
+
+class AddPetForm extends StatefulWidget {
+  final Function(Pet) addPetHandler;
+
+  const AddPetForm({Key key, this.addPetHandler}) : super(key: key);
+
+  @override
+  _AddPetFormState createState() => _AddPetFormState();
+}
+
+class _AddPetFormState extends State<AddPetForm> {
+  final StorageService _storageService = services.get<StorageService>();
+  final AuthService _authService = services.get<AuthService>();
+  File _image;
+  final _formKey = GlobalKey<FormState>();
+  final nameController = TextEditingController();
+  final typeController = TextEditingController();
+  final biographyController = TextEditingController();
+  final pictureUrlController = TextEditingController();
+  final ageController = TextEditingController();
+
+  Future getImage() async {
+    final imageSource = await AppDialogs.chooseImageSource(context);
+
+    if(imageSource != null) {
+      final file = await ImagePicker.pickImage(source: imageSource);
+      if(file != null) {
+        setState(() => _image = file);
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    nameController.dispose();
+    typeController.dispose();
+    biographyController.dispose();
+    pictureUrlController.dispose();
+    ageController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Pet petObject = ModalRoute.of(context).settings.arguments;
+
+    return Form(
+      key: _formKey,
+      child: Padding(
+        padding: EdgeInsets.symmetric(horizontal: 20.0),
+        child: Column(
+          children: <Widget>[
+            Row(mainAxisAlignment: MainAxisAlignment.center, children: <Widget>[
```

Comment:
Can you extract this `Row` widget to a custom widget so that we can reuse it when we have to show a profile pic?